docs/rfcs/0010-add-state-based-navigation.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: page
+title: Add State-based navigation
+permalink: /rfc/0010-add-state-based-navigation
+---
+
+* Start date: 2023-10-17
+* RFC PR: [4126](https://github.com/mozilla-mobile/firefox-android/pull/4126)
+
+## Summary
+
+Following the acceptance of [0009 - Remove Interactors and Controllers](0009-remove-interactors-and-controllers), Fenix should have a method of navigation that is tied to the `lib-state` model to  provide a method of handling navigation side-effects that is consistent with architectural goals.
```

Comment:
> that is consistent with architectural goals.
nit: Is there some doc that you can link here to outline the architectural goals?

Reply:
Good call, updated!

Reply:
Nice 👌